src/commands/react.ts

```diff
@@ -0,0 +1,168 @@
+
+import * as Discord from 'discord.js';
+
+import { Inject, AutoWired, Singleton } from 'typescript-ioc';
+
+import { ICommand, ICommandArgs, ICommandResult } from '../interfaces';
+import { EmojiService } from '../services/EmojiService';
+import { Logger } from '../services/Logger';
+import { ReactService } from '../services/ReactService';
+import { ErrorMessageService } from '../services/ErrorMessageService';
+import { Channel, DMChannel, GroupDMChannel, Message, TextChannel } from 'discord.js';
+
+@Singleton
+@AutoWired
+export class ReactCommand implements ICommand {
+
+  help = 'React to the message immediately above the command or react to a specific' +
+    'message with a message-id\n' +
+    '**Usage**\n' +
+    'react message-id emoji-name - reacts to a specific message\n' +
+    'react emoji-name - reacts to the above message\n' +
+    '**Supported Servers**\n' +
+    'PMMM: Magia Record\n' +
+    '/r/MadokaMagica\n' +
+    'Madoka\'s Disciples\n' +
+    'Magireco Emotes for Discord Whales\n' +
+    'and other partnered Emoji servers';
+  aliases = ['react', 'r'];
+
+  @Inject private logger: Logger;
+  @Inject private emojiService: EmojiService;
+  @Inject private reactService: ReactService;
+  @Inject private errorMessageService: ErrorMessageService;
+
+  async execute(cmdArgs: ICommandArgs): Promise<ICommandResult> {
+    const {message, args} = cmdArgs;
+
+    const commandArgs = args.split(' ');
+
+    // Args should only be length 1 or 2
+    if (commandArgs.length !== 1 && commandArgs.length !== 2) {
+      // Error, number of arguments doesn't match
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  You must either include an emoji name or a message-id and then an emoji name!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Argument error occurred', result: {success: false}};
+    }
+
+    if (commandArgs.length === 1) {
+      return await this.reactNow(commandArgs[0], message);
+    } else {
+      return await this.reactToMessage(commandArgs[0], commandArgs[1], message);
+    }
+  }
+
+  /**
+   * React to the message immediately above
+   * @param emojiName - The emoji name
+   * @param message - The command message
+   */
+  async reactNow(emojiName: string, message: Discord.Message): Promise<ICommandResult> {
+    const emoji = this.emojiService.getEmojiByName(emojiName);
+
+    if (!emoji) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find that emoji!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return { resultString: 'Could not find the emoji', result: {success: false}};
```

Comment:
Make sure you run the linter. I think this is an error due to whitespace.

Reply:
Just ran it and looks like this is fine.  Got an error for consecutive blank lines in IGirlData though: 
![image](https://user-images.githubusercontent.com/41814814/61597950-e38be480-abe4-11e9-8aa6-0a2609590418.png)

That file wasn't changed in this, so I'll leave that for a different pull request